### DISCUSSION TITLE
can't connect a 14.xx flashed device to my home AP

### DISCUSSION THREAD

**Discussion by Mark-Muc** [arendst/Tasmota#21484] [Q&A] [answered]
Hello . some question about 14.0 (core3)
If I flash a new device (using webinstaller), it dosen't matter witch version i took .. also uploded own compiled .factory.bin.
I can flash witout problems. but if i then (also after reboot) try to connect wo my home AP (fritz-Box) via the wenbflasher or connect to AP the tasmotadevice opens an got to 192.168.4.1 in my browsers (edge&chrome) only the Tasmotasign is visible ... nothing else.
in Webflasher only "configure wi-Fi" is visible and it run for ever.
I habe attached some screenshots zo display. I tried it now for 3 days!!
I can flash a final version of Tasmota (13.4.0) and all works fine .. both Webflasher & AP connect (192.168.4.1)
The i can insall via firmware update a 14.xxx version, also if i update it twice, first time it run into "safe boot" second try works.
Someone know about this problem??
Thx for help
Mark_Muc
some pics:

![chrome](https://github.com/arendst/Tasmota/assets/67410847/9e792520-08e7-4607-9d3b-a3cb1090c4a5)
![webflaher](https://github.com/arendst/Tasmota/assets/67410847/837c1ccd-58be-4a1d-a116-ba222ca4ee72)

**Comment by sfromis** [comment]
Depending on your hardware setup, one possible complication is if your power source (like an USB/serial adapter) cannot supply enough power for normal operation of Tasmota.

**Comment by Mark-Muc** [comment]
thx for the answer!!
I forgot to mention.
Same setup on device (esp32) USB/ Seriel adapert .. everything the same .. but ONLY the .bin file changes.
it is connected in to an USB 3.0 HUB with external power supply, should have  enough power
it spans a owen AP .. like normal!!
Working is 13.4.0 final
NOT working: 14.xxxxx 
regards
Mark_Muc

**Comment by Jason2866** [comment]
Which MCU? What Os do you have? (Windows, Linux MacOS)

**Comment by Mark-Muc** [comment]
i run it on windows 11 ...Az Delivery Esp32 device with serial converter on board and boot switch (d0 -> gnd)
https://www.google.com/amp/s/www.az-delivery.de/a/s/products/esp32-developmentboard

________________________________

**Comment by Jason2866** [comment]
So nothing special. This behaviour is very typical for power issues. Since the underlying base (IDF) has completely changed there is the possibility that the device needs more power now (when starting).
It is not safe to say power supply is good "just" when the previous version works.
Try a different power supply and a different board. Both can be "bad"

**Reply by TD-er** [reply]
And also try a different USB cable as some connectors (either on board or cable) have pins which are a bit short or making bad contact for other reasons.
So changing the USB cable can also make a difference as not all cables are created equal and some do show wear by plugging/unplugging over time.

**Comment by Mark-Muc** [comment]
Thanks fot the help here.
I found now the problem itself on this 2 issues.
Issue 1: not connecting to my AP!
It works, if i use the final 14.0. versions to flash, it works with simply tasmota 2 and also with tasmota32 + Bluetooth. both connected to the hub and to the USB 3.0 (the blue one) direkt in the PC. Ruining Windows 11 on it.
It doesn't work (that was may problem) it I tried a version later den 14.0.0 (final).. i tested also an all ports and also upload an own compiled version!! no connection possible. 
Simply workaround!
flash with 14.0.0 final 
connect to my AP. 
updated to the desired (own compiled) version 2 times!!. Don't know why 2 times. First time it reboots to "fail safe"  mode. Second time update it worked and BLE devices will be found and displayed.

issue 2: lagginging very slow / reboot etc.
If i aplay my own Tmeplate:
`{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[0,0,0,0,32,576,0,0,0,0,0,33,0,0,416,225,0,0,228,224,0,0,0,0,0,0,0,0,0,0,226,0,227,0,0,0],"FLAG":0,"BASE":1}
`
it starts to lagging. 
Template copied from a running and working device (13.4) final.
I divined some own relay's but their is no function behind it, just to get powerstates to handle in a rule.
here are my used config:
![image](https://github.com/arendst/Tasmota/assets/67410847/77a9d5a0-83b7-4dc7-95a1-619ec7ab32bb)

here th org. template:
`{"NAME":"Sonoff SwitchMan 2C","GPIO":[0,0,0,0,32,576,0,0,0,0,0,33,0,0,416,225,0,0,0,224,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
`
so i applied the template to a fresh installed 14.0.0 (final) version from here:
[https://templates.blakadder.com/sonoff_SwitchMan_M5-2C.html](url)
asap it lagged and hung.
 
That's it so far!
Mark_Muc

**Comment by Jason2866** [comment]
Please try without using the backup. Configure the device "by hand".

**Reply by Mark-Muc** [reply]
Jason:
I did and it becomes very strange. I did it step by step . first 2 relays .. etc.
It working all fine und no lagging. 
so i downloaded the Template and compared it with the template of the old one:

```
new: (configured by hand) Version Tasmota 14.0.0.1 (TasmoCompiler-esp32generic) von Theo Arends
{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[1,1,1,1,32,576,1,1,1,1,1,33,1,1,416,225,0,1,228,224,0,1,1,1,0,0,0,0,1,1,226,1,227,0,0,1],"FLAG":0,"BASE":1}

old: used with Tasmota 13.4.0(TasmoCompiler-esp32generic) von Theo Arends
{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[0,0,0,0,32,576,0,0,0,0,0,33,0,0,416,225,0,0,228,224,0,0,0,0,0,0,0,0,0,0,226,0,227,0,0,0],"FLAG":0,"BASE":1}

```
very strange .. the new one is working 

up to now it is working an no lagging. also configured as shuter .. autoexec.bat is uploaded and working.

Regards
Mark_Muc

**Comment by Jason2866** [comment]
@Mark-Muc Please try latest development branch. I have updated the Arduino framework.

**Reply by Mark-Muc** [reply]
i will to it tomorrw .. Thanks u

**Reply by Mark-Muc** [reply]
Good Morning!!
So i compiled an new version in dev brache and played with it.
As in the versions bevor I can't connect it to my AP (Frirz.Box). after i flashed the final (14.0.0 final) I was able to connect to the Wi-Fi.
it is the same behavior like bevor.
I flashed to 14.0.0 and then try to updated the device via Firmware update. Same as bevor .. i selected the new .bin (not .factory.bin) it reboots came up in FAILSAFE mode (like bevor) a scond try of Firmewate update then updates the device und cam up (same procedure like before).

I also connected to serieal port in the webfalsher, so i can controle the devive a bit (i.e. reset 5).
So i played back my settings (old ones) and et voila no lagging anymore and looked good for the first time.
Tried to upload my autoecec.bat .. it was done but no reacton of the device i had to switch it on and off. Ok it showed up.
some lagging . .so i did "reset 5" in console and applied the new (yesterday) mad settings (.dmp) .. it lloks better . fileupload worked as expecteted. 
Then swiching of MQTT (other configuration) the device was lagging und never showed up. switched it of and on, several times, no reaction. even seriel connetion (webflasher) i cant access the device, I have to reflash the device!!

I think there's a lot wrong with Tasmota at the moment.
Regards
Mark_Muc
I can provide the settings in privat!!

**Reply by TD-er** [reply]
About the MQTT...
I also have seen some reports for ESPEasy where some users have suddenly issues with MQTT.
I have not yet pinpointed what MQTT broker or broker config they are using.
Maybe some special MQTT broker included with some image like for Domoticz ???

**Reply by Mark-Muc** [reply]
Thx TD-er

**Comment by Mark-Muc** [comment]
i think it has to do with my Templates. Also if i confgured it manual.
Simply to repoduce:
First craeat a 32bit ESP device. flash it with 14.0.0 .. connect it to the network.
go to the IP and siply apply one of the Templates:
```
new: (configured by hand) Version Tasmota 14.0.0.1 (TasmoCompiler-esp32generic) von Theo Arends
{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[1,1,1,1,32,576,1,1,1,1,1,33,1,1,416,225,0,1,228,224,0,1,1,1,0,0,0,0,1,1,226,1,227,0,0,1],"FLAG":0,"BASE":1}

old: used with Tasmota 13.4.0(TasmoCompiler-esp32generic) von Theo Arends
{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[0,0,0,0,32,576,0,0,0,0,0,33,0,0,416,225,0,0,228,224,0,0,0,0,0,0,0,0,0,0,226,0,227,0,0,0],"FLAG":0,"BASE":1}

```
after the reboot it hangs / lagging after some clics in the menues .. 

Regards
Mark_Muc
PS. running the templates on Tasmota (13.4.0) bevor Core 3 .. no problem!!

**Reply by sfromis** [reply]
Functionally those two templates are 100% identical. The only difference is that unused pins are configured as "None" vs "User", which **only** affects whether they show up on the "Configure Module" page. The purpose of that is to flag which pins are "unavailable to tinkering".

If both fails with the new, and not with the old Tasmota version, it has nothing to do with your templates.

**Reply by Mark-Muc** [reply]
yes i saw that they only differ in 0 / 1 and the last 0 / 1, 
sure both fail on "core 3".
Mark_Muc

**Comment by Jason2866** [comment]
No idea what could be the reason for the problem you encounter.
Can't reproduce.
To get a clue, do not compile in Bluetooth. Still the same?

**Reply by Mark-Muc** [reply]
yes i just took the first Tasmota Version in webfalsher .. no Bluetooth!
if i click several times . disable MQTT / aktivate MQTT .. and it hags. also try to uplaod a smal file ...
Mark_Muc

**Comment by Mark-Muc** [comment]
Hello and good morning!!
I now think i fount the problem!!
I again now setup a new device.
installed tasmota32 with bluetooth.
updated to my own compiled (newest sources). It is the same:
First it loads the file but then nothing happens, afer reboot the device it cam up in safeboot mode and THEN i can update to my .bin.
Nothing changed on that.
uploded my file: autoecec.bat
```
Mi32Option6 1
MI32Page 10
MI32Block A4C138465A57
MI32Block E7764500F442
BLEAlias 150311116C66=LYWSD02-1 A4C138AF4A0B=Temp&Hum(LYWSD03MMC) E72E0050A0CC=1.OG_Bad A4C138465A57=Pavillion_innen A4C1382D29C3=1.OG_Arbeit_2 A4C1389C4D2A=Thermostat
shutteropen

```
again i have ti power off/on the device, but then it works,

Then I applied the template:
`{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[1,1,1,1,32,576,1,1,1,1,1,33,1,1,416,225,0,1,228,224,0,1,1,1,0,0,0,0,1,1,226,1,227,0,0,1],"FLAG":0,"BASE":1}
`
still working fine!!
The problem now came after i set MQTTT to Emulation: `x Hue Bridge ...`
![image](https://github.com/arendst/Tasmota/assets/67410847/9839226d-549f-45e6-87f5-276493cee362)
 after the reboot it start lagging!!
after a lot of reboots I was able to switch it back to: `x none`
ad it works normal speed.
Pls Jason can u give it a last try?
Thank u
Mark_Muc

**Reply by TD-er** [reply]
Can you also see in the logs of the MQTT broker whether the ESP device makes a connection and/or if it reconnects?
Or do you still need to configure a MQTT broker somewhere else?

**Comment by Mark-Muc** [comment]
The MQTT broker itselfe has to be defined in Menue: MQTT-Settings

**Comment by Jason2866** [comment]
Hue Emulation does work for me, but enabling fires a WDT when enabling. No lagging after it is enabled.
@s-hadinger There is a timeout WDT in Hue Emulation

**Reply by s-hadinger** [reply]
I don't see anything in Hue Emulation that should cause a pause, especially because Hue Emulation relies only on non-blocking UDP. This is very strange

**Comment by Jason2866** [comment]
@s-hadinger Mhh, indeed. @Mark-Muc we have disabled the WDT. Does you have the HUE issue still with latest development build?

**Comment by Mark-Muc** [comment]
i'll test tomorrow ..

so I did .. built a new version (14.0.2) updated to my device (software update). working so far!
but after set MQTT - Emulation to` x Hue Bridge ...` it lagged like before!
Regards Mark_Muc

. now i will falsh it complet new with the` .factory.bin` and let u know here!

**Comment by Mark-Muc** [comment]
so i now set up with webfasher my own compiled .factory.bin , connected it to my Wi-Fi .. working fine . was able to set MQTT-Emulation to Hue Bridge ... working like expected .. the nI applied my Template:
`{"NAME":"Sonoff SwitchMan 2C LED (button)","GPIO":[1,1,1,1,32,576,1,1,1,1,1,33,1,1,416,225,0,1,228,224,0,1,1,1,0,0,0,0,1,1,226,1,227,0,0,1],"FLAG":0,"BASE":1}`
if i apülied the template before set the Emulation to Hue Bright .. it works fine .. then set the Emulation Hue Bright -> laging.
Might be that 6 defined relais are too much ... seems to be!

and it hangs . can only reboot it by power off / on but as far as it boots up it hangs / laging.

**Comment by Mark-Muc** [comment]
some remarks:
Setup via webflaher / own factory.bin now working an connecting to the Wi-Fi .. 
also the update process now working .. 
but upload a file (i.e auteoxec.bat) ended in I have to reboot the device!!

**Reply by arendst** [reply]
I think we have an issue with UDP, used in emulation Wemo/Hue on core3 as used in v14.0.0 FYI see https://github.com/arendst/Tasmota/discussions/21488#discussioncomment-9577337

To be continued.

**Comment by arendst** [comment, marked as answer]
Fixed in v14.0.0.2 See #21470

**Comment by Mark-Muc** [comment]
THX to all, fixing the existing Problems!!

Feedback:
I installed it (14.0.0.3); self compiled on the following Devices:
```
Software:
Tasmota 14.0.0.3 (TasmoCompiler-esp32generic) von Theo Arends
Tasmota Version	14.0.0.3 (TasmoCompiler-esp32generic)

```
```
Devices:
Sonoff SwitchMan 2C (ESP Chip ID 13228868 (ESP32-D0WD-V3 v3.0)) 2 devices
Sonoff Dual R3 v2 (ESP Chip ID	11180448 (ESP32-D0WD-V3 v3.0)) 5 devices
Shelly Plus 2PM PCB v0.1.9 (ESP Chip ID	8864892 (ESP32-U4WDH-D v3.0)) 4 devices
DYI based on Wemos LOLIN32 V1.0.0 (ESP Chip ID 15715344 (ESP32-D0WDQ6 v1.0)) 1 devices
 - GPIO2 AM2301
 - GPIO4 Relay 2
 - GPIO12 Relay 3
 - GPIO13 Relay 4
 - GPIO16 Switch_n 1
 - GPIO16 Relay 1
 - GPIO21 I2C SDA
 - GPIO22 I2C SDA
{"I2CScan":"Device(s) found on bus1 at 0x23"} -> BH1750

```
I did an update on all 12  devices, different compiler switches!!
_Everything went smooth and without any problem._
again thanks to all for fixing it so fast!!
Regards 
Mark_Muc
